src/GiraffeGenerator.Core/OpenApi.fs

```diff
@@ -232,19 +357,51 @@ let parse (doc: OpenApiDocument): Api =
                     let methodName = normalizeName (string method)
                     let opName = normalizeName op.OperationId
                     
+                    let pathParameters = Option.ofObj path.Parameters |> Option.map Seq.toArray
+                    let operationParameters = Option.ofObj op.Parameters |> Option.map Seq.toArray
+                    let allParameters =
+                        Option.map2 Array.append pathParameters operationParameters
+                        |> Option.orElse operationParameters
+                        |> Option.orElse pathParameters
                     let parameters =
-                        Option.ofObj op.Parameters
-                        |> Option.filter (fun x -> x.Count > 0)
-                        |> Option.map (fun parameters ->
-                            TypeSchema.Parse(methodName + opName, parameters))
+                        allParameters
+                        |> Option.filter (fun x -> x.Length > 0)
+                        |> Option.map
+                            (fun parameters ->
+                                parameters
+                                |> Seq.groupBy (fun x -> Option.ofNullable x.In |> Option.defaultValue ParameterLocation.Query |> PayloadLocation.FromParameterLocation)
+                                |> Seq.map (fun (location, parameters) ->
+                                    location, TypeSchema.Parse(methodName + opName + location.ToString(), parameters))
+                            )
+                    let bodyParameters =
+                        op.RequestBody
+                        |> Option.ofObj
+                        |> Option.map
+                            (
+                                fun rb ->
+                                    rb.Content
+                                    |> Seq.map
+                                        (
+                                           fun (KeyValue(mt, body)) ->
+                                               let mediaType = parseMediaType mt
+                                               let name = opName + methodName + "Body" + mediaType.ToString()
+                                               let schema = TypeSchema.Parse(name, body.Schema)
+                                               Body mediaType, schema
+                                        )
+                            )
```

Comment:
It's probably Fantomas, but indentation here is pretty generous :D
Let's reformat this code a little bit?

Reply:
It's a manual formatting here. What would you like this code to be formatted like?

Reply:
e.g.
```fsharp
|> Option.map ^fun rb ->
  rb.Content
  |> Seq.map ^fun (KeyValue(mt, body)) ->
    let mediaType = parseMediaType mt
    ..
```

Reply:
Reformatted it with Egyptian-style curly braces because cap operator seems to have a priority issue in this case (it's usage here produces type mismatch compiler error)